tools/bumper-bot/parser/parser.go

```diff
@@ -0,0 +1,48 @@
+package parser
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+
+	"gopkg.in/yaml.v2"
+)
+
+type Component struct {
+	Name        string `yaml:"name"`
+	Url         string `yaml:"url"`
+	Commit      string `yaml:"commit"`
+	Release_tag string `yaml:"release_tag"`
+}
+
+type ComponentsConfig struct {
+	Components []Component
+}
+
+func ParseComponentsYaml(componentsYaml string) ([]Component, error) {
+	config := ComponentsConfig{}
+
+	fmt.Println("opening components config", componentsYaml)
+	componentsData, err := ioutil.ReadFile(componentsYaml)
+
+	fmt.Println("Unmarshaling components config", componentsYaml)
+	err = yaml.Unmarshal(componentsData, &config)
+	if err != nil {
+		return config.Components, errors.Wrapf(err, "failed to Unmarshal %s", componentsYaml)
+	}
+
+	return config.Components, nil
+}
+
+func UpdateComponentsYaml(components []Component) error {
```

Comment:
Use ComponentsConfig all the way

Reply:
I now return map, so it's a totally different struct.